### PR TITLE
Modify the searchMedia() function in Api.js.

### PR DESCRIPTION
**Pull request by dmitrizagidulin**
Addresses issue https://github.com/xrchat/kxr-dev/issues/19

### CODE REVIEW

**Comment by dmitrizagidulin** [comment]
(reviewing)

File: src/api/Api.js [hunk 1]
```diff
@@ -357,7 +370,9 @@ export default class Project extends EventEmitter {
     }
 
     if (params.query) {
-      searchParams.set("q", params.query);
+      //checking BLOCK_SEARCH_TERMS
+      if (this.searchTermFilteringBlacklist(params.query)) searchParams.set("q", "");
```
Comment (dmitrizagidulin) [inline src/api/Api.js]: What does setting `q` to an empty string do? Does that return an empty set of results back to the user?

File: src/ui/projects/usePaginatedSearch.js [hunk 2]
```diff
@@ -2,6 +2,7 @@ import { useState, useCallback, useEffect, useRef } from "react";
 import { useFetch } from "use-http";
 
 export default function usePaginatedSearch(path, queryParams, options = {}) {
+  console.log(queryParams);
```
Comment (dmitrizagidulin) [inline src/ui/projects/usePaginatedSearch.js]: Please remove stray console.log

**Comment by dmitrizagidulin** [comment]
@igorgark1593 
Right now, the function `searchTermFilteringBlacklist()` checks if the _entire_ query is blacklisted. Don't forget though (and this is mentioned in the original issue, "Break the params.query string into words"), that if there are multiple words, you need to check for each one.

So, if my blocked list is `['cat', 'dog']`, and I search for the words `orange cat`, that query should return 0 results, because `cat` is a blocked word.

File: src/api/Api.js (hunk 1 above)
Comment (igorgark1593) [reply src/api/Api.js]: run Spoke and see what happens in the request

File: src/ui/projects/usePaginatedSearch.js (hunk 2 above)
Comment (igorgark1593) [reply src/ui/projects/usePaginatedSearch.js]: ok

**Comment by igorgark1593** [comment]
Right now, the function searchTermFilteringBlacklist() checks if the entire query is blacklisted. Don't forget though (and this is mentioned in the original issue, "Break the params.query string into words"), that if there are multiple words, you need to check for each one.

So, if my blocked list is ['cat', 'dog'], and I search for the words orange cat, that query should return 0 results, because cat is a blocked word.

File: src/api/Api.js (hunk 1 above)
Comment (dmitrizagidulin) [reply src/api/Api.js]: @igorgark1593 Or you can just tell me :)

**Pull request closed without merging**